File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/PutRoomRequest.java
package com.ll.hotel.domain.hotel.room.dto;

import com.ll.hotel.domain.hotel.room.type.BedTypeNumber;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Objects;
import java.util.Set;
import org.hibernate.validator.constraints.Length;

public record PutRoomRequest(

        @NotBlank @Length(min = 2, max = 30) String roomName,

        @NotNull @Min(value = 0) Integer roomNumber,

        @NotNull Integer basePrice,

        @NotNull @Min(value = 1) Integer standardNumber,

        @NotNull @Min(value = 1) Integer maxNumber,

        @NotNull BedTypeNumber bedTypeNumber,

        @NotBlank String roomStatus,

        List<String> deleteImageUrls,

        List<String> imageExtensions,

        Set<String> roomOptions
) {
    public PutRoomRequest {
        deleteImageUrls = Objects.requireNonNullElse(deleteImageUrls, new ArrayList<>());
        imageExtensions = Objects.requireNonNullElse(imageExtensions, new ArrayList<>());
        roomOptions = Objects.requireNonNullElse(roomOptions, new HashSet<>());
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/PutRoomRequest.kt
package com.ll.hotel.domain.hotel.room.dto

import com.ll.hotel.domain.hotel.room.type.BedTypeNumber
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

data class PutRoomRequest(
    @field:NotBlank
    @field:Length(min = 2, max = 30)
    val roomName: String,

    @field:Min(value = 0)
    val roomNumber: Int,

    val basePrice: Int,

    @field:Min(value = 1)
    val standardNumber: Int,

    @field:Min(value = 1)
    val maxNumber: Int,

    val bedTypeNumber: BedTypeNumber,

    @field:NotBlank
    val roomStatus: String,

    val deleteImageUrls: List<String> = emptyList(),

    val imageExtensions: List<String> = emptyList(),

    val roomOptions: Set<String> = emptySet()
)